python/tests/scripts/b1_equal_chain.metta

```diff
@@ -1,43 +1,66 @@
-; Equality sign is specially treated by the interpreter
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; The equality sign is specially treated by the interpreter.
+
+; The evaluation of any (expr) in MeTTa is done via
+; constructing the following equality query:
+; (match &self (= (expr) $r) $r)
+; The result $r is recursively evaluated in the same way,
+; until the match is empty (when the equality match is
+; empty, the expression is evaluated to itself).
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 (= (I $x) $x)
 (= ((K $x) $y) $x) ; Note that `(K $x)` is allowed here
 (= (K $x $y) $x)
 (= (S $x $y $z) ($x $z ($y $z)))
 
-; Evaluation of (expr) is done via constructing
-; the equality query (match &self (= (expr) $r) $r)
-; The result $r is then evaluated in the same way
-; until the match is not empty (if equality match is
-; empty, the expression is evaluated to itself)
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ; `assertEqual` compares (sets of) results of evaluation
 ; of two expressions (both expressions can have empty
 ; sets of results, so `assertEqualToResult` is safer)
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 !(assertEqual
   (S K K x)
   x)
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ; Here, we use `assertEqualToResult` to make sure
 ; that the expression is evaluated to itself
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 !(assertEqualToResult
    (expression without known equalities)
   ((expression without known equalities)))
 
 
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ; Another example
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 (= (Add $x Z) $x)
 (= (Add $x (S $y)) (Add (S $x) $y))
 
 !(assertEqual
   (Add (S Z) (S Z))
   (S (S Z)))
 
-; In the current implementation, the child expression will be reduced
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; In the current implementation, child expressions
+; are also reduced
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 !(assertEqual
   (Something? (Add (S Z) (S Z)))
   (Something? (S (S Z))))
 
-; Note that the same variable can be used multiple time on the left
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+; Note that the same variable can be used multiple times on the left
 ; side of expression, and the "function" can be "non-total"
+; (a "non-total" function can be defined only for some inputs;
+; is not required to cover all possible inputs)
```

Comment:
These two lines are redundant saying similar things. Moreover, if the function is not total, it really is not defined for all inputs (not just "can" or "not required"). If it is defined for all inputs, it is total.

Reply:
I meant the text in parenthesis as an explanation of what is a "total-function". Mostly because I didn't know what it means, but I changed wording to fix your suggestion:

Reply:
I'm not sure if this terminology is precise, but it can be improved later if needed.